### DISCUSSION TITLE
Preserve component state when navigating back

### DISCUSSION THREAD

**Discussion by niddu85** [nuxt/nuxt#19905] [Questions]
Hello,
I am looking for a way to save the state of a component ONLY on browser back button is pushed.
EG: in a search form within a page the user enters search parameters and launches search. In the same page search results will appear. The user clicks on a search result to open details (navigating to a new page), but then he wants to go back and expects to find the search parameters set previously

I'm trying to play with the keepalive flag but i was able to make it work only setting it in nuxt.config.ts for the whole application, because using definePageMeta in a specific page doesn't work.
The problem is this will save the sate of the component forever, even if i land on the same page after few consequient clicks.
I was also able to set the keepalive flag within a navigation middleware in defineNuxtRouteMiddleware setting eg to.meta.keepalive = true but here I didn't find any way to detect if the back key was pressed

Is there a way to do that without saving the state manually and manually restoring it?

thanks in advance

**Comment by mahdiboomeri** [comment]
You can use a keyed `<KeepAlive>` component. Whenever the key changes, the state will reset.

**Reply by niddu85** [reply]
good suggestion, I just have to understand how to "invalidate" the state when the page is navigated directly instead of being navigated with "back"
